src/System.Runtime.Extensions/tests/System/Environment.ProcessorCount.cs

```diff
@@ -0,0 +1,64 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Runtime.InteropServices;
+using Xunit;
+
+namespace System.Runtime.Extensions.Tests
+{
+    public class EnvironmentProcessorCount
+    {
+        [Fact]
+        public void ProcessorCountTest()
+        {
+            //arrange
+            int expected;
+
+            if(Interop.IsWindows)
+            {
+                SYSTEM_INFO sysInfo = new SYSTEM_INFO();
+                SYSTEM_INFO.GetSystemInfo(ref sysInfo);
+                expected = sysInfo.dwNumberOfProcessors;
+            }
+            else
+            {
+                int _SC_NPROCESSORS_ONLN = Interop.IsOSX ? 58 : 84;
```

Comment:
Since this is now specific to a particular flavor of Unix, consider looking up the right value for FreeBSD and including it.  Alternatively, to make sure this is flagged for when the FreeBSD port team starts going through the se tests, you could assert that either IsOSX or IsLinux.